3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune/PCAWrapper.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Xamarin.Forms;
+
+namespace activedirectoryxamarinintune
+{
+    /// <summary>
+    /// This is a wrapper for PCA. It is singleton and can be utilized by both application and the MAM callback
+    /// </summary>
+    public class PCAWrapper
+    {
+
+        /// <summary>
+        /// This is the singleton used by consumers
+        /// </summary>
+        static public PCAWrapper Instance { get; }
+
+        internal IPublicClientApplication PCA { get; }
+
+        // This event coordinates between different callbacks
+        public static ManualResetEvent MAMRegsiteredEvent { get; } = new ManualResetEvent(false);
```

Comment:
nit: I generally prefer to use SemaphoreSlim in these kind of situations because it has async semantics built in and plays better with async, but this works as well.

Reply:
May be next time.